backend/tests/end_to_end/infrastructures/apps/load/test_admin_panel.py

```diff
@@ -0,0 +1,231 @@
+    assert "data" in form.errors
+
+
+def test_admin_form_raises_validation_error_when_age_is_not_a_number():
+    # Given
+    form_data = {
+        "data": {"full_name": "John Doe", "age": "eleven", "is_satisfied": True}
+    }
+
+    # When
+    form = admin.DataAdminForm(data={"data": form_data["data"]})
+
+    # Then
+    assert not form.is_valid()
+    assert "data" in form.errors
+
+
+def test_admin_form_raises_validation_error_when_name_is_not_a_string():
+    # Given
+    form_data = {"data": {"full_name": 12345, "age": 11, "is_satisfied": True}}
+
+    # When
+    form = admin.DataAdminForm(data={"data": form_data["data"]})
+
+    # Then
+    assert not form.is_valid()
+    assert "data" in form.errors
+
+
+def test_admin_form_raises_validation_error_when_name_is_satisfied_is_not_bool():
+    # Given
+    form_data = {"data": {"full_name": "John Doe", "age": 11, "is_satisfied": "True"}}
+
+    # When
+    form = admin.DataAdminForm(data={"data": form_data["data"]})
+
+    # Then
+    assert not form.is_valid()
+    assert "data" in form.errors
+
+
+def test_export_to_excel(
+    authenticated_client: Client,
+):
+    # Given
+    row_number = 5
+    for _ in range(row_number):
+        model_factories.DataFactory.create()
+
+    # When
+    response = authenticated_client.post(
+        get_url(
+            path_name="admin:load_data_export",
+        ),
+        {
+            "timestamp_from": "",
+            "timestamp_to": "",
+            "format": "0",  # CSV format
+            "resource": "",
+            "_export": "Export",
+        },
+        follow=True,
+    )
+
+    # Then
+    assert response.status_code == HTTPStatus.OK
+
+    rows = list(csv.reader(io.StringIO(response.content.decode("utf-8"))))
+
+    assert len(rows) == row_number + 1  # row with columns included
+    expected_columns = ["id", "full_name", "is_satisfied", "age", "created_at"]
+    assert all(column in rows[0] for column in expected_columns)
+
+
+def test_export_to_excel_correct_row_exported_for_given_time_range(
+    authenticated_client: Client,
+):
+    # Given
+    with freeze_time("2023-06-01T00:00:00"):
+        model_factories.DataFactory.create()
+
+    with freeze_time("2024-06-01T00:00:00"):
+        data: Data = model_factories.DataFactory.create()
+
+    with freeze_time("2023-06-01T00:00:00"):
+        model_factories.DataFactory.create()
+
+    # When
+    response = authenticated_client.post(
+        get_url(
+            path_name="admin:load_data_export",
+        ),
+        {
+            "timestamp_from": "2023-12-31T00:00:00",
+            "timestamp_to": "2025-01-01T00:00:00",
+            "format": "0",  # CSV format
+            "resource": "",
+            "_export": "Export",
+        },
+        follow=True,
+    )
+
+    # Then
+    assert response.status_code == HTTPStatus.OK
+    rows = list(csv.reader(io.StringIO(response.content.decode("utf-8"))))
+
+    assert len(rows) == 2  # row with columns included
```

Comment:
Always use all()

Reply:
how exactly do you want to check number of rows using all()?

Reply:
Never-mind, i just wanted to check something else, but I noticed that we are already checking that.